bittensor/_neuron/text/core_server/run.py

```diff
@@ -0,0 +1,347 @@
+            if pubkey in timecheck.keys():
+                prev_time = timecheck[pubkey]
+                if current_time - prev_time >= timedelta(seconds=config.neuron.blacklist.time):
+                    timecheck[pubkey] = current_time
+                    return False
+                else:
+                    timecheck[pubkey] = current_time
+                    raise Exception('blacklist')
+            else:
+                timecheck[pubkey] = current_time
+                return False
+
+        # Black list or not
+        try:
+            #registration_check()
+
+            #stake_check()
+
+            #validator_check()
+            
+            return False
+
+        except Exception as e:
+            return True
+
+    def backward_callback(inputs_x:torch.FloatTensor, grads_dy:torch.FloatTensor, synapses=[] ):
+        """
+            The default forward callback when no callback is attached: Is used to call specific synapse functions
+
+            Args:
+                inputs_x (:obj:`torch.FloatTensor`, `required`): 
+                    The inputs that will be passed to the synapse functions
+                
+                synapses (:obj: list of bittensor.proto.SynapseArgs, 'Optional')
+                    The proto message that contains additional args for individual synapse functions
+
+            Returns:
+                response_tensors: (:obj: list of bittensor.proto.Tensor, `required`): 
+                    serialized tensor response from the nucleus call or None.
+                response_codes: (:obj: list of bittensor.proto.ReturnCode, `required`)
+                    return code associated with forward call i.e. Success of Timeout.
+                response_messages: (:obj: list of strings, `required`)
+                    return message associated with synapse call
+        """
+        # --- initialize response variables --- 
+        response_tensors = []
+        response_codes = []
+        response_messages = []
+        
+        if not config.neuron.remote_train:
+            return response_tensors, response_codes, response_messages
+
+
+        # --- calling attached synapses ---
+        with mutex and torch.enable_grad() and torch.autograd.set_detect_anomaly(True):
+            for index, synapse in enumerate(synapses):
+                try:
+                    if synapse.synapse_type in axon.synapse_callbacks and axon.synapse_callbacks[synapse.synapse_type] != None:
+                        model_output, response_tensor = axon.synapse_callbacks[synapse.synapse_type](inputs_x[index], synapse)
+                        grads_dy_norm = grads_dy[index]/(grads_dy[index].sum() + 0.00001)
+                        torch.autograd.backward (
+                            tensors = [ response_tensor ],
+                            grad_tensors = [ grads_dy_norm ],
+                            retain_graph=True
+                        )                        
```

Comment:
Similar to the advanced server, we apply the gradients and do one large step, correct?

Reply:
yea, that's right